### PR TITLE
Add mergify

### PR DESCRIPTION
**Pull request by wirednkod**
Added mergify following the example of Smoldot.

Im not sure if we should have autoapproval for dependabot:
```
  - name: automatic approval for Dependabot pull requests
    conditions:
      - author~=^dependabot(|-preview)\[bot\]$
    actions:
      review:
        type: APPROVE
        message: Automatically approving dependabot
```

Opinions @tomaka ?

### CODE REVIEW

File: .github/mergify.yml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+queue_rules:
+  - name: default
+    conditions:
+      - check-success=all-ci
```
Comment (tomaka) [inline .github/mergify.yml:4]: This "all-ci" is the name of the CI pipeline which has to succeed for the PR to be mergeable

File: .github/mergify.yml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+queue_rules:
+  - name: default
+    conditions:
+      - check-success=build
```
Comment (tomaka) [inline .github/mergify.yml:4]: Let's depend on `all`
https://github.com/paritytech/substrate-connect/pull/774

Comment (wirednkod) [reply .github/mergify.yml:4]: That was my first approach but we have only 1 for now, and when a second is added, then this can be added... But sure

Comment (tomaka) [reply .github/mergify.yml:4]: We already have three: `build (14.x)`, `build (16.x)` and `build (17.x)`

**Review by tomaka — APPROVED** [review]

**Pull request merged**